cmd_wait_ready.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/lightningnetwork/lnd/lncfg"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/signal"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	connectionRetryInterval = time.Millisecond * 250
+)
+
+type waitReadyCommand struct {
+	RPCServer string        `long:"rpcserver" description:"The host:port of lnd's RPC listener"`
+	Timeout   time.Duration `long:"timeout" description:"The maximum time we'll wait for lnd to become ready; 0 means wait forever"`
+}
+
+func newWaitReadyCommand() *waitReadyCommand {
+	return &waitReadyCommand{
+		RPCServer: defaultRPCServer,
```

Comment:
nit: (optional) we could default the timeout here to forever and then there's no need to check it in `Execute`

Reply:
Good idea, but then the help text will show: `(default: 2562047h47m16.854775807s)` which isn't really very nice IMO. So I'd rather have the check below.